From 164c33e29bfe06d99c58c1ee73dfd6f31dc4c663 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 30 Sep 2023 10:18:54 +1000
Subject: [PATCH] d3dx9: [HACK] D3DXIntersect - Assign hit to FALSE

This functions needs to be implement to make "Humanity Asset" to work correctly.

Added to stop the Infinite loop - Doesn't make the game playable.
---
 dlls/d3dx9_36/mesh.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/d3dx9_36/mesh.c b/dlls/d3dx9_36/mesh.c
index 822b2d7d410..fdc642c4e0c 100644
--- a/dlls/d3dx9_36/mesh.c
+++ b/dlls/d3dx9_36/mesh.c
@@ -7645,6 +7645,8 @@ HRESULT WINAPI D3DXIntersect(ID3DXBaseMesh *mesh, const D3DXVECTOR3 *ray_pos, co
     FIXME("mesh %p, ray_pos %p, ray_dir %p, hit %p, face_index %p, u %p, v %p, distance %p, all_hits %p, "
             "count_of_hits %p stub!\n", mesh, ray_pos, ray_dir, hit, face_index, u, v, distance, all_hits, count_of_hits);
 
+    *hit = FALSE;
+
     return E_NOTIMPL;
 }
 
-- 
2.42.0

